Gordon360/Services/HousingService.cs

```diff
@@ -531,4 +557,228 @@ public bool ChangeApplicationDateSubmitted(int applicationID)
+                var existActiveApplicant = context.Applicant.FirstOrDefault(x => (x.Applicant1 == e) && (x.Active == 1));
+                if (existActiveApplicant != null)
+                {
+                    existActiveApplicant.Active = 0;
+                }
+                var student = context.Student.FirstOrDefault(x => x.Email == e);
+                if (student == null)
+                {
+                    throw new ResourceNotFoundException() { ExceptionMessage = "The applicant cannot be found." };
+                }
+                if (gender != student.Gender)
+                {
+                    throw new BadInputException() { ExceptionMessage = "The applicants are not of the same gender." };
+                }
+                gender = student.Gender;
+                var newApplicant = new Applicant
+                {
+                    ApplicationID = application_id,
+                    Applicant1 = e,
+                    Active = 1
+                }; ;
+                await context.Applicant.AddAsync(newApplicant);
+            }
+        }
+        context.SaveChanges();
+    }
+
+    /// <summary>
+    /// update the information of preferred halls
+    /// </summary>
+    /// <param name="username"> The username for the user who complete the aplication form </param>
+    /// <param name="application_id"> The ID of this application </param>
+    /// <param name="hallList"> A list of the preferred halls </param>
+    public void UpdatePreferredHall(string username, string application_id, string[] hallList)
+    {
+        var hallPreferences = hallList.Select((hall, index) => new PreferredHall { ApplicationID = application_id, Rank = index + 1, HallName = hall });
+        context.PreferredHall.AddRange(hallPreferences);
+        context.SaveChanges();
+    }
+
+    /// <summary>
+    /// update the information of preferred halls
+    /// </summary>
+    /// <param name="username"> The username for the user who complete the aplication form </param>
+    /// <param name="application_id"> The ID of this application </param>
+    /// <param name="preferenceList"> A list of the preference </param>
+    public void UpdatePreference(string username, string application_id, string[] preferenceList)
+    {
+        var preferences = preferenceList.Select((preference) => new Preference { ApplicationID = application_id, Preference1 = preference });
+        context.Preference.AddRange(preferences);
+        context.SaveChanges();
+    }
+
+    /// <summary>
+    /// update the information of due date
+    /// </summary>
+    /// <param name="dueDate"> The due date of the application </param>
+    public async Task UpdateDueDateAsync(string dueDate)
+    {
+        // This code here is not working correctly. It cannot save the changes.
+        var myDueDate = context.Config.FirstOrDefault(d => d.Key == "housing_lottery_due_date");
+        myDueDate.Value = dueDate;
+        await context.SaveChangesAsync();
+    }
```

Comment:
The code here to update the row of `housing_lottery_due_date` is not working. It cannot save the new due date. When I troubleshoot with break points, I found that it was able to update the `myDueDate.Value` on line 646, but seems not be able to save it on line 647. By the way, it works if I drop the row and insert the new row, but it doesn't work if I update the value like this.

Reply:
Do you have an error message that you see?